test/test_code_action.vader

```diff
@@ -261,22 +328,37 @@ Execute(Cursor will not move when it is before text change):
   let g:test.changes = g:test.create_change(2, 3, 2, 8, 'value2')
 
   call setpos('.', [0, 1, 1, 0])
-  call ale#code_action#HandleCodeAction(g:test.changes, {'should_save': 1})
+  call ale#code_action#HandleCodeAction(g:test.changes, {
+  \ 'should_save': 1,
+  \ 'conn_id': 'test_conn',
+  \})
   AssertEqual [1, 1], getpos('.')[1:2]
 
+  " make tests at the end work
+  let g:notified_changes = []
+
   call setpos('.', [0, 2, 2, 0])
-  call ale#code_action#HandleCodeAction(g:test.changes, {'should_save': 1})
+  call ale#code_action#HandleCodeAction(g:test.changes, {
+  \ 'should_save': 1,
+  \ 'conn_id': 'test_conn',
+  \})
   AssertEqual [2, 2], getpos('.')[1:2]
 
 # ====C====
 Execute(Cursor column will move to the change end when cursor between start/end):
   let g:test.changes = g:test.create_change(2, 3, 2, 8, 'value2')
 
   for r in range(3, 8)
+    " make tests at the end work
+    let g:notified_changes = []
```

Comment:
It might be better to assert the contents of `g:notified_changes` explicitly/deliberately in some tests (where we test for the notification) rather than to pollute most tests with workarounds for an assertion in the test teardown, I think.

Reply:
You're right, added more and better assertions and removed these resets.